### PR TITLE
Support multiple files - manually tested

### LINKED ISSUE #42: Support lock generation from 1..* files
Thanks for conda-lock! Starting to using it in anger!

A use case I'm currently working on solving with jinja, but would consider PRing (eventually, free time, yadda yadda): generating locks based on multiple "inherited" environment files. Consider:

```yaml
# env-base.yml
channels:
  - conda-forge
dependencies:
  - python =3.7
  # a cast of thousands...
```

```yaml
# env-cpu.yml
channels:
  - conda-forge
dependencies:
  - pytorch-cpu
```

```yaml
# env-gpu.yml
channels:
  - pytorch
  - conda-forge
dependencies:
  - pytorch
```

then be able to:

```bash
conda-lock -f env-cpu.yml -f env-base.yml -p osx-64 -p linux-64 --prefix cpu-
conda-lock -f env-gpu.yml -f env-base.yml -p osx-64 -p linux-64 --prefix gpu-
```

... and get
```bash
cpu-conda-linux-64.lock
cpu-conda-osx-64.lock
gpu-conda-linux-64.lock
gpu-conda-osx-64.lock
```

of course, it gets freakier with env-specific yml, etc. as well as extending this to other environment sources, e.g. #39, #40, #41...

### PR DESCRIPTION
**Pull request by munali**
Closes #42

### CODE REVIEW

File: conda_lock/conda_lock.py [hunk 1]
```diff
@@ -343,16 +347,32 @@ def parser():
...
+
+def aggregate_lock_specs(lock_specs: List[LockSpecification]) -> LockSpecification:
+    # union the dependencies
+    specs = list(set(chain.from_iterable([lock_spec.specs for lock_spec in lock_specs])))
+
+    # pick the first non-empty channel
+    channels = next((lock_spec.channels for lock_spec in lock_specs if lock_spec.channels), [])
+
+    # pick the first non-empty platform
+    platform = next((lock_spec.platform for lock_spec in lock_specs if lock_spec.platform), "")
```
Comment (mariusvniekerk) [inline conda_lock/conda_lock.py:373]: platforms should all be the same

File: conda_lock/conda_lock.py [hunk 2]
```diff
@@ -343,16 +347,32 @@ def parser():
...
+        if src_file.name == "meta.yaml":
+            desired_envs.append(parse_meta_yaml_file(src_file, platform, include_dev_dependencies))
+        elif src_file.name == "pyproject.toml":
+            desired_envs.append(parse_pyproject_toml(src_file, platform, include_dev_dependencies))
+        else:
+            desired_envs.append(parse_environment_file(src_file, platform))
+    return desired_envs
+
+
+def aggregate_lock_specs(lock_specs: List[LockSpecification]) -> LockSpecification:
```
Comment (mariusvniekerk) [inline conda_lock/conda_lock.py:365]: Can you add a test for this function?

File: conda_lock/conda_lock.py [hunk 3]
```diff
@@ -343,16 +347,32 @@ def parser():
...
-        desired_env = parse_meta_yaml_file(src_file, platform, include_dev_dependencies)
-    elif src_file.name == "pyproject.toml":
-        desired_env = parse_pyproject_toml(src_file, platform, include_dev_dependencies)
-    else:
-        desired_env = parse_environment_file(src_file, platform)
-    return desired_env
+def parse_source_files(
+    src_files: List[pathlib.Path], platform: str, include_dev_dependencies: bool
+) -> List[LockSpecification]:
+    # todo: any problem if src_files are a mixture of meta, toml or enviroment files?
```
Comment (mariusvniekerk) [inline conda_lock/conda_lock.py:353]: shouldn't be a problem

**Comment by scopatz** [comment]
Thanks for putting this in @munali!

File: conda_lock/conda_lock.py (hunk 2 above)
Comment (munali) [reply conda_lock/conda_lock.py:368]: added

**Review by mariusvniekerk — APPROVED** [review]

**Pull request merged**